packages/core/src/model/GraphProcessor.ts

```diff
@@ -869,10 +874,12 @@ export class GraphProcessor {
     if (!inputsReady) {
       return;
     }
-    this.#emitter.emit(
-      'trace',
-      `Node ${node.title} has required inputs nodes: ${inputNodes.map((n) => n.title).join(', ')}`,
-    );
+    if(this.#includeTrace){
+      this.#emitter.emit(
+        'trace',
+        `Node ${node.title} has required inputs nodes: ${inputNodes.map((n) => n.title).join(', ')}`,
+      );
+    }
```

Comment:
Also we should consider adding a small util to wrap this so future calls to emitting trace don't miss the check